src/symbol/projection.js

```diff
@@ -323,6 +320,7 @@ function placeGlyphsAlongLine(symbol, fontSize, flip, keepUpright, posMatrix, la
 
         if (keepUpright && !flip) {
             const orientationChange = requiresOrientationChange(symbol, firstPoint, lastPoint, aspectRatio);
+            symbol.needsFlipping = orientationChange ? FlipDecision.flipRequired : FlipDecision.flipNotRequired;
```

Comment:
Do we need a further check like `orientationChange && orientationChange.needsFlpping`

Reply:
Though you're right, to protect against possible other cases, yes, it's safest that way.
```suggestion
            symbol.needsFlipping = orientationChange && orientationChange.needsFlipping ? FlipDecision.flipRequired : FlipDecision.flipNotRequired;
```